app/services/search/wider_suggestions_builder.rb

```diff
@@ -15,16 +15,15 @@ def initial_radius
   end
 
   def suggestions
-    RADIUS_OPTIONS
-      .select { |r| r > initial_radius }
-      .map { |radius| [radius.to_s, wider_results(radius)] }
-      .uniq(&:second)
-      .reject { |options| options.second.zero? }
+    @suggestions ||= RADIUS_OPTIONS.select { |r| r > initial_radius }
+                                   .map { |radius| [radius.to_s, wider_results_count(radius)] }
+                                   .uniq(&:second)
+                                   .reject { |options| options.second.zero? }
```

Comment:
Costly computation. Computes the search between 1 and 9 times (depending on initial_radius).